CodeGeneration/Sources/Utils/SyntaxBuildableChild.swift

```diff
@@ -18,6 +18,7 @@ import SyntaxSupport
 /// given token kind.
 public enum SyntaxOrTokenNodeKind: Hashable {
   case node(kind: SyntaxNodeKind)
+  case nodeChoices(syntaxType: TypeSyntax)
```

Comment:
I was wondering that the `syntaxType` is here. Seems like it’s essentially `Child.syntaxChoicesType`. Could you add a comment for that? Also: Why didn’t we need this for the non-raw syntax types?

Reply:
There's a lack of a centralized way to do things in code generation. In non-raw syntax types, there's a special handling in place,
```swift
let childType: TypeSyntax =
  child.kind.isNodeChoicesEmpty ? child.syntaxNodeKind.syntaxType : child.syntaxChoicesType
let type = child.isOptional ? TypeSyntax("\(childType)?") : TypeSyntax("\(childType)")
```
I'm working on the consolidation of common logic across raw and non-raw types, maybe I've fallen into the trap of scope creep...

Reply:
I see. It would be good to migrate that to use similar logic as the raw nodes but I agree that it can be a follow-up PR.

I was a little concerned about the new `nodeChoices` case because the previous `node` and `token` case used associated value that will reference types available at the global level in swift-syntax, while the new case references an elements type that is only available as a nested type. But I think it makes sense with your explanation. Could you add some documentation for it though, so it’s clear what `syntaxType` will be?